.github/workflows/test.yml

```
@@ -0,0 +1,37 @@
+name: Java tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

I'm realizing the change I suggested here is causing the unit tests not to run on this PR either since it's a PR into `dev` instead of `main`.  Maybe we should make this:

```suggestion
    branches: [ ** ]
```

So all PRs are tested. I still don't think we need to test every commit push though.